sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_create_options.hpp

```diff
@@ -58,4 +60,206 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
     std::unordered_map<std::string, std::string> Tags;
   };
 
+  /**
+   * @brief The properties needed to create an Elliptic Curve key.
+   *
+   */
+  class CreateEcKeyOptions : public CreateKeyOptions {
+  private:
+    std::string m_name;
+    bool m_hardwareProtected;
+    JsonWebKeyType m_keyType;
+
+  public:
+    /**
+     * @brief Gets or sets the elliptic curve name.
+     *
+     * @remark See #KeyCurveName for possible values.
+     *
+     * @remark If null, the service default is used.
+     *
+     */
+    Azure::Core::Nullable<KeyCurveName> CurveName;
+
+    /**
+     * @brief Create a Ec Key Options object.
+     *
+     * @param name Name of the key to create.
+     * @param hardwareProtected `true` to create hardware-protected key in a hardware security
+     * module (HSM). The default is false to create a software key.
+     */
+    explicit CreateEcKeyOptions(std::string const& name, bool hardwareProtected = false)
+        : m_hardwareProtected(hardwareProtected)
+    {
+      if (name.empty())
+      {
+        throw std::invalid_argument("The name can't be empty");
+      }
+      m_name = name;
+      if (hardwareProtected)
+      {
+        m_keyType = JsonWebKeyType::EcHsm;
+      }
+      else
+      {
+        m_keyType = JsonWebKeyType::Ec;
+      }
+    }
+
+    /**
+     * @brief Gets the name of the key to create.
+     *
+     */
+    std::string const& GetName() const { return m_name; };
+
+    /**
+     * @brief Gets the key type to create, including Ec and EcHsm.
+     *
+     */
+    JsonWebKeyType GetKeyType() const { return m_keyType; }
+
+    /**
+     * @brief Gets a value indicating whether to create a hardware-protected key in a hardware
+     * security module (HSM).
+     *
+     */
+    bool GetHardwareProtected() const { return m_hardwareProtected; }
+  };
+
+  /**
+   * @brief The properties needed to create an RSA key.
+   *
+   */
+  class CreateRsaKeyOptions : public CreateKeyOptions {
+  private:
+    std::string m_name;
+    bool m_hardwareProtected;
+    JsonWebKeyType m_keyType;
+
+  public:
+    /**
+     * @brief Gets or sets the key size in bits, such as 2048, 3072, or 4096.
+     *
+     * @remark If null, the service default is used.
+     *
+     */
+    Azure::Core::Nullable<uint64_t> KeySize;
```

Comment:
Why 64-bit ints here and below? Both of these values are small. Is this just idiomatic for the C++ SDK like `int` is for C# in general (when size really doesn't matter much)?

Reply:
This is because the functions from the standard library uses `size_t` for things like list size, count, etc. This is to be agnostic to the platform (x86, x64 or other).
Using uint32_t or something smaller would require an static cast to `size_t` when using the type to compare against any `size_t` type.  But uint64_t is directly compatible with size_t

Reply:
In this case it wouldn't matter, but you can otherwise always upcast unsigned types. You can't always downcast signed or unsigned types, so for 32-bit code you'd have to downcast bitlengths from uint64 to size_t when it's a uint32. Values here should never be that high, but I'm concerned about the general case where it could matter.